tests/test_object_states.py

```diff
@@ -0,0 +1,692 @@
+    dishtowel.set_position([0, 0., 0.63])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert bowl.states[Inside].get_value(bottom_cabinet)
+    assert dishtowel.states[Inside].get_value(bottom_cabinet)
+
+    bowl.set_position([10., 10., 1.])
+    dishtowel.set_position([20., 20., 1.])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert not bowl.states[Inside].get_value(bottom_cabinet)
+    assert not dishtowel.states[Inside].get_value(bottom_cabinet)
+
+    assert bowl.states[Inside].set_value(bottom_cabinet, True)
+    assert dishtowel.states[Inside].set_value(bottom_cabinet, True)
+
+    with pytest.raises(NotImplementedError):
+        bowl.states[OnTop].set_value(bottom_cabinet, False)
+
+@og_test
+def test_under():
+    breakfast_table = og.sim.scene.object_registry("name", "breakfast_table")
+    bowl = og.sim.scene.object_registry("name", "bowl")
+    dishtowel = og.sim.scene.object_registry("name", "dishtowel")
+
+    breakfast_table.set_position([0., 0., 0.53])
+    bowl.set_position([0., 0., 0.04])
+    dishtowel.set_position([0.3, 0., 0.02])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert bowl.states[Under].get_value(breakfast_table)
+    assert dishtowel.states[Under].get_value(breakfast_table)
+
+    bowl.set_position([10., 10., 1.])
+    dishtowel.set_position([20., 20., 1.])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert not bowl.states[Under].get_value(breakfast_table)
+    assert not dishtowel.states[Under].get_value(breakfast_table)
+
+    assert bowl.states[Under].set_value(breakfast_table, True)
+    assert dishtowel.states[Under].set_value(breakfast_table, True)
+
+    with pytest.raises(NotImplementedError):
+        bowl.states[Under].set_value(breakfast_table, False)
+
+@og_test
+def test_touching():
+    breakfast_table = og.sim.scene.object_registry("name", "breakfast_table")
+    bowl = og.sim.scene.object_registry("name", "bowl")
+    dishtowel = og.sim.scene.object_registry("name", "dishtowel")
+
+    breakfast_table.set_position([0., 0., 0.53])
+    bowl.set_position([0., 0., 0.7])
+    dishtowel.set_position([0.5, 0., 0.67])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert bowl.states[Touching].get_value(breakfast_table)
+    assert breakfast_table.states[Touching].get_value(bowl)
+    assert dishtowel.states[Touching].get_value(breakfast_table)
+    assert breakfast_table.states[Touching].get_value(dishtowel)
+
+    bowl.set_position([10., 10., 1.])
+    dishtowel.set_position([20., 20., 1.])
+
+    for _ in range(5):
+        og.sim.step()
+
+    assert not bowl.states[Touching].get_value(breakfast_table)
+    assert not breakfast_table.states[Touching].get_value(bowl)
+    assert not dishtowel.states[Touching].get_value(breakfast_table)
+    assert not breakfast_table.states[Touching].get_value(dishtowel)
+
+    with pytest.raises(NotImplementedError):
+        bowl.states[Touching].set_value(breakfast_table, False)
```

Comment:
should we also check for True in this case?

Reply:
(same goes for other tests, e.g. NextTo that don't support setting True nor False)

Reply:
I changed it to `bowl.states[Touching].set_value(breakfast_table, None)` to test for both True and False. Hopefully it is sufficient.